src/components/molecules/searchBar/SearchBar.js

```diff
@@ -0,0 +1,22 @@
+import React, {useRef} from 'react'
+import Button from '../../atoms/button/Button'
+import InputText from '../../atoms/textInput/InputText'
+import {getRepoList} from '../../../service/getRepoList'
+
+export default function SearchBar(props) {
+
+  const searchName = useRef();
+
+  const getRepoInfo = async(userName) => {
+    const result = await getRepoList(userName);
+    props.setRepoList(result);
+    props.setUserName(userName);
+  }
+  
+  return (
+    <div className="SearchBar">
+      <InputText placeholder={"검색할 Github User 또는 Organization의 이름을 입력해주세요!"} searchName={searchName} />
+      <Button name="Search" onClick={() => getRepoInfo(searchName.current.value)}/>
+    </div>

```

Comment:
ref를 사용하는거 말고도, 이런 방법으로도 가능할 것 같습니다 (손코딩 하느라 제데로 친지 헷갈리네요)
```suggestion
const [searchName, setSearchName] = useState("");

  const getRepoInfo = async(userName) => {
    const result = await getRepoList(userName);
    props.setRepoList(result);
    props.setUserName(userName);
  }
  
  const handleChange = useCallback((e)=> {
    setUserName(e.target.value);
  }, [setUserName]);
  
  return (
    <div className="SearchBar">
      <InputText
          placeholder={"검색할 Github User 또는 Organization의 이름을 입력해주세요!"} 
          onChange={handleChange} value={userName}}/>
      <Button name="Search" onClick={() => getRepoInfo(userName)}/>
    </div>

```

Reply:
ref 말고 useCallback...! 기억하겠습니다 ㅎㅎㅎ 감사합니다!

Reply:
`<Button name="Search" onClick={() => getRepoInfo(userName)}/>`
요 부분에서 핸들러가 인라인 함수로 들어가게 되면 렌더링 마다 새로운 함수로 인식하고 생성하게 되어서,
`onClick={onSearchHandler}` 와 같이 하는 것도 좋을 것 같습니다..!